src/index.test.ts

```diff
@@ -1184,15 +1184,20 @@ test('Should call isEnabled event when impressionData is false and impressionDat
     });
 
     client.on(EVENTS.IMPRESSION, (event: any) => {
-        expect(event.featureName).toBe('impression');
-        expect(event.eventType).toBe('isEnabled');
-        expect(event.impressionData).toBe(undefined);
-        client.stop();
-        done();
+        try {
```

Comment:
Why the try/catch here? If it's needed, a try/finally might be better here

Reply:
Good point! I explained it in the commit, but not in the PR, I see. It's explained more thoroughly in the [callback section of Jest's testing asynchronous code](https://jestjs.io/docs/asynchronous#callbacks), but in short:

If you don't have the try/catch and one of the expectations fail, then the `done` callback is never called (because the function short circuits). While this _does_ mean that the test fails (this is good), it fails with a message saying that it exceeded the set timeout and not because it failed an expectation (this is bad).

To instead have the correct error message reported, you must catch the error and pass it to the `done` callback. This will make the test fail **and** report the correct error message.

Now regarding using finally, I _did_ consider that but ended up not doing it for the following reason:

Calling the `done` callback requires the error for it to fail. This can't be done in a finally block because you don't have access to the error. In theory, you could probably declare an `error` variable that you set to undefined and then assign in the `catch` block, but that doesn't feel any cleaner to me. I'd be happy to take suggestions, though 😄

Reply:
A follow-up question, then, could be: why didn't I add try/catch clauses to **all the other tests** that also use the `done` callback? I thought about it, but didn't think it'd be worth it. The tests still fail, even if their error messages are confusing, and the diff would've been much bigger and harder to understand if I _had_ gone and done that. 

However, I could do it now if you think it's sensible? 💁🏼

Reply:
Right, gosh that is arcane but definitely nice to have!

I think it's fine as is. This is effectively a 4 character change to source code that's already quite a large diff. It's nice to have commits self contained so if you're keen you're welcome to do it in a separate PR but I think this one stands as it is without adding more to it.